prac_06/car.py

```diff
@@ -0,0 +1,35 @@
+"""CP1404/CP5632 Practical - Car class example."""
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, name, fuel=0):
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.name = name
+        self.fuel = fuel
+        self._odometer = 0
+
+    def __str__(self):
+        return f"{self.name}, fuel={self.fuel}, odometer={self._odometer}"
+
+    def add_fuel(self, amount):
+        """Add amount to the car's fuel."""
+        self.fuel += amount
+
+    def drive(self, distance):
+        """Drive the car a given distance.
+
+        Drive given distance if car has enough fuel
+        or drive until fuel runs out return the distance actually driven.
+        """
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
```

Comment:
could this be simplified by making it so self.fuel -= distance is outside of the if statement? As setting self.fuel to distance means taking it from itself will be equal to 0 anyway